hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -766,15 +767,19 @@ private void logIgnoredCompletion(@Nullable Throwable failure, JobStatus status)
 
     private void completeVertices(@Nullable Throwable failure) {
         if (vertices != null) {
-            for (Vertex vertex : vertices) {
-                try {
-                    ProcessorMetaSupplier metaSupplier = vertex.getMetaSupplier();
-                    Util.doWithClassLoader(metaSupplier.getClass().getClassLoader(), () -> metaSupplier.close(failure));
-                } catch (Throwable e) {
-                    logger.severe(mc.jobIdString()
-                            + " encountered an exception in ProcessorMetaSupplier.close(), ignoring it", e);
+            JobExecutionService jobExecutionService = mc.getJetServiceBackend().getJobExecutionService();
+            ClassLoader jobCL = jobExecutionService.getClassLoader(mc.jobConfig(), mc.jobId());
+            doWithClassLoader(jobCL, () -> {
```

Comment:
We need to set the jobCL for the `PMS#close` for all the vertices, so I put it outside of the iteration. Then there might be a particular vertex with processorCL, so that's set inside the loop (if the classloader is null then the `doWithClassLoader` is a no-op.